src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,72 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> interfaceImplementationMap = Map
+            .of(ProductService.class, ProductServiceImpl.class,
+                    ProductParser.class, ProductParserImpl.class,
+                    FileReaderService.class, FileReaderServiceImpl.class);
+
+    private Injector() {
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isInterface()) {
```

Comment:
The method `getInstance` currently throws an exception if the provided class is not an interface. This is problematic because it assumes that only interfaces can be injected. Consider modifying this logic to handle concrete classes as well.